internal/cli/clients/client.go

```diff
@@ -939,6 +949,30 @@ func (c *EpinioClient) Push(app string, source string) error {
 		return errors.Wrap(err, "waiting for app failed")
 	}
 
+	if len(services) > 0 {
+		c.ui.Note().Msg("Binding Services")
+
+		// Application is up, bind the services.
+		// This will restart the application.
+		// TODO: See #347 for future work
```

Comment:
I'm not sure we need #347 to fix this. We can simply send over the desired services as part of the `upload` request and let the server do the bindings. This way the client does as little as possible. That's the approach I took in #368 to handle the `instances` parameter. 
Since we are going to have conflicts with #368, let's see if we can improve this code there. Let's keep it as it is for now.

Reply:
Thanks. Hm ... `upload` provides the sources as tarball, in the request body ... I guess if the framework does the correct thing for us when we add additional parameters like services and #instances, then it should work fine. I guess the `upload` is then more of an `app create` API. And Mario's work removing the event listener and web hook, etc. then gets a separate `app start` API.